src/miner.cpp

```diff
@@ -158,7 +180,12 @@ bool SolveProofOfStake(CBlock* pblock, CBlockIndex* pindexPrev, CWallet* pwallet
     }
     // Stake found
     pblock->nTime = nTxNewTime;
-    pblock->vtx[0].vout[0].SetEmpty();
+    CMutableTransaction emptyTx;
+    emptyTx.vin.resize(1);
+    emptyTx.vin[0].scriptSig = CScript() << pindexPrev->nHeight + 1 << OP_0;
+    emptyTx.vout.resize(1);
+    emptyTx.vout[0].SetEmpty();
+    pblock->vtx.emplace_back(CTransaction(emptyTx));
     pblock->vtx.emplace_back(CTransaction(txCoinStake));
```

Comment:
Nit:  can pass directly the mutable objects, `emptyTx` and `txCoinStake`, to `emplace_back()`, so the `CTransaction` objects are constructed in place.

Reply:
updated.